src/components/OrdersWidget/index.tsx

```diff
@@ -238,85 +248,57 @@ const OrdersWidget: React.FC = () => {
                 {['active', 'liquidity'].includes(selectedTab) ? 'Cancel' : 'Delete'} {markedForDeletion.size} orders
               </ButtonWithIcon>
             </div>
-            {/* PENDING ORDERS */}
-            {pendingShownOrdersCount ? (
-              <div>
-                <h3>Pending Orders</h3>
-                <div className="ordersContainer">
-                  <CardTable
-                    $columns="minmax(13.625rem, 1.3fr) repeat(2, minmax(6.2rem, 0.6fr)) minmax(5.5rem, 0.6fr)"
-                    $cellSeparation="0.2rem"
-                    $rowSeparation="0.6rem"
-                  >
-                    <thead>
-                      <tr>
-                        <th>Order details</th>
-                        <th>Unfilled amount</th>
-                        <th>Account balance</th>
-                        <th>Expires</th>
-                      </tr>
-                    </thead>
-                    <tbody>
-                      {pendingOrders.map((order: PendingTxObj) => (
-                        <OrderRow
-                          key={Math.random()}
-                          order={order}
-                          networkId={networkId}
-                          isOverBalance={false}
-                          pending
+            {ordersCount > 0 ? (
+              <div className="ordersContainer">
+                <CardTable
+                  // $columns="minmax(2rem, min-content) minmax(13.625rem, 1fr) repeat(2, minmax(6.2rem, 0.6fr)) minmax(5.5rem, 1fr)"
+                  $columns="minmax(2rem,.4fr)  minmax(11rem,1fr)  minmax(11rem,1.3fr)  minmax(5rem,.9fr)  minmax(auto,1.4fr)"
+                  // $cellSeparation="0 .5rem;"
+                  $rowSeparation="0"
+                >
+                  <thead>
+                    <tr>
+                      <th className="checked">
+                        <input
+                          type="checkbox"
+                          onChange={toggleSelectAll}
+                          checked={markedForDeletion.size === displayedOrders.length}
                           disabled={deleting}
-                          isPendingOrder
                         />
-                      ))}
-                    </tbody>
-                  </CardTable>
-                </div>
+                      </th>
+                      <th>Limit price</th>
+                      <th className="filled">Filled / Total</th>
+                      <th>Expires</th>
+                      <th className="status">Status</th>
+                    </tr>
+                  </thead>
+                  <tbody>
+                    {displayedPendingOrders.map((order: PendingTxObj) => (
+                      <OrderRow
+                        key={Math.random()}
```

Comment:
Please, don't use random keys. OrderRow will remount each time.
Why not `order.id` or `order.txHash`. `order.id` here should not interfere with same key on L289, they are different arrays, if you are worried about that

Reply:
Haven't done this, it was there already. This PR was unification.

I didn't see it, I agree it's super bad practice. I'll address it in separate PR

Reply:
let's um, not use git blame here

Reply:
think u just discovered the flashing pending orders bug :joy: :joy: god daaamn

Reply:
Gilt is shared among reviewers too :) 
Nice that Dima spotted it now!